Automattic/wp-calypso
# A4A: unify form validation

Resolves https://github.com/Automattic/automattic-for-agencies-dev/issues/765

## Proposed Changes

A proposition to create HOC to unify form validation within A4A. It wraps the logic inside HOC and triggered `onFocus` and `onBlur` events. 

I've added this validation to few fields on Agency Details form. Please check these out. 
Note: I haven't removed old Hook validation for now.

@cleacos I don't remove old validation, so currently both will be in place. We can remove it in a follow up PR not to overload this one.

<img width="720" alt="Screenshot 2024-07-05 at 1 24 22 PM" src="https://github.com/Automattic/wp-calypso/assets/60262784/83a76f85-11bb-4df3-a630-ab5385621778">

## Why are these changes being made?



## Testing Instructions



- Navigate to partner directory folder and check behavior of Agency expertise and Agency details forms

## Pre-merge Checklist
